packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -393,53 +407,53 @@ BlockOption.propTypes = {
   blockSelected: PropTypes.string.isRequired,
 };
 
-const ListButton = ({ icon, format, label, disabled }) => {
-  const editor = useSlate();
+/**
+ *
+ * @param {import('slate').Node} node
+ * @returns boolean
+ */
+const isListNode = (node) => {
+  return !Editor.isEditor(node) && SlateElement.isElement(node) && node.type === 'list';
+};
 
-  /**
-   *
-   * @param {import('slate').Node} node
-   * @returns boolean
-   */
-  const isListNode = (node) => {
-    return !Editor.isEditor(node) && SlateElement.isElement(node) && node.type === 'list';
-  };
+const isListActive = (editor, format) => {
+  const { selection } = editor;
 
-  const isListActive = () => {
-    const { selection } = editor;
+  if (!selection) return false;
 
-    if (!selection) return false;
+  const [match] = Array.from(
+    Editor.nodes(editor, {
+      at: Editor.unhangRange(editor, selection),
+      match: (node) => isListNode(node) && node.format === format,
```

Comment:
it looks like we could use the matchNode function from the store to replace both this and the isListNode functions. Maybe ListButton could receive the whole block object from the store as a prop since it uses most attributes